internal/quaiapi/quai_api.go

```diff
@@ -409,7 +409,7 @@ func (s *PublicBlockChainQuaiAPI) EstimateGas(ctx context.Context, args Transact
 	if nodeCtx != common.ZONE_CTX {
 		return 0, errors.New("estimateGas can only called in a zone chain")
 	}
-	bNrOrHash := rpc.BlockNumberOrHashWithNumber(rpc.PendingBlockNumber)
+	bNrOrHash := rpc.BlockNumberOrHashWithNumber(rpc.LatestBlockNumber)
```

Comment:
This violates the expected behavior of this API. This API is supposed to return gas estimate for the current pending block.

It would certainly be better to provide the estimate on pending. Is there no way to do that?

If not, then we also need to change the code comment above and probably some other API documentation.